src/submission/validation-clinical/therapy.ts

```diff
@@ -270,3 +293,41 @@ export function getTreatment(
 
 	return treatment;
 }
+
+const validateDrugTreatmentFields = (
+	therapyRecord: DeepReadonly<SubmittedClinicalRecord>,
+	errors: SubmissionValidationError[],
+) => {
+	const { drug_name, drug_rxnormcui, drug_database, drug_term, drug_id } = therapyRecord;
+	const isRxNorm = Boolean(drug_name && drug_rxnormcui);
+	const isDrugDb = Boolean(drug_database && drug_id && drug_term);
+
+	if (!isRxNorm && isDrugDb) {
+		// No RxNorm, All Drug DB Fields, no further validation
+		return;
+	} else if (!isRxNorm && !isDrugDb) {
+		// Not RxNorm, but Not All Drug Fields -> Error
+		const drugFields = { drug_database, drug_term, drug_id };
+		const missingFields: ClinicalInfo[] = [];
+		for (const [drugKey, drugValue] of Object.entries(drugFields)) {
+			if (!drugValue) {
+				const data = { [drugKey]: drugValue };
+				missingFields.push(data);
+			}
+		}
+
+		missingFields.forEach((field) => {
+			const [drugKey] = Object.entries(field)[0];
```

Comment:
why not just name this `fieldName` instead of new var assignment?

Reply:
See above